components/FlexibleHeader/src/private/MDCFlexibleHeaderMinMaxHeight.h

```diff
@@ -0,0 +1,147 @@
+ area.
+
+ If neither the minimumHeight nor maximumHeight are modified, this class will dynamically update
+ both properties as the top safe area inset changes with the assumption that it completely owns
+ both property's values. This covers the majority case of clients that are using the default
+ values. For all other clients that set explicit minimumHeight and maximumHeight values, we
+ recommend that they disable minMaxHeightIncludesSafeArea rather than attempt to implement the
+ correct top safe area logic themselves.
+ */
+@interface MDCFlexibleHeaderMinMaxHeight : NSObject
+
+#pragma mark Initializing a min/max height object
+
+/**
+ Initializes this instance with a given top safe area instance.
+ */
+- (nonnull instancetype)initWithTopSafeArea:(nonnull MDCFlexibleHeaderTopSafeArea *)topSafeArea
+    NS_DESIGNATED_INITIALIZER;
+
+#pragma mark Configuring the minimum and maximum height
+
+/**
+ See MDCFlexibleHeaderView.h for complete documentation of this property.
+ */
+@property(nonatomic) CGFloat minimumHeight;
+
+/**
+ See MDCFlexibleHeaderView.h for complete documentation of this property.
+ */
+@property(nonatomic) CGFloat maximumHeight;
+
+#pragma mark Calculating the minimum and maximum height with/without the top safe area
+
+/**
+ Returns minimum height including the top safe area amount.
+ */
+- (CGFloat)minimumHeightWithTopSafeArea;
+
+/**
+ Returns maximum height including the top safe area amount.
+ */
+- (CGFloat)maximumHeightWithTopSafeArea;
+
+/**
+ Returns maximum height without the top safe area amount.
+ */
+- (CGFloat)maximumHeightWithoutTopSafeArea;
+
+#pragma mark Delegating changes in minimum and maximum height
+
+/**
+ The delegate may react to changes in the minimumHeight and maximumHeight.
+ */
+@property(nonatomic, weak) id<MDCFlexibleHeaderMinMaxHeightDelegate> delegate;
+
+/**
+ Unavailable. Use initWithTopSafeArea: instead.
+ */
+- (instancetype)init NS_UNAVAILABLE;
+
+@end
+
+#pragma mark APIs that will be deprecated
+
+@interface MDCFlexibleHeaderMinMaxHeight ()
+
+/**
+ See MDCFlexibleHeaderView.h for complete documentation of this property.
+
+ This property will eventually be disabled by default, deprecated, and then removed.
+ */
+@property(nonatomic) BOOL minMaxHeightIncludesSafeArea;
+
+/**
+ Informs the receiver that it should update the minimumHeight and maximumHeight.
+
+ Does nothing if minMaxHeightIncludesSafeArea is disabled.
+ This property can be removed once minMaxHeightIncludesSafeArea is removed.
+ */
+- (void)recalculateMinMaxHeight;
+
+@end
+
+/**
+ The delegate protocol through which MDCFlexibleHeaderMinMaxHeight communicates changes in the
+ minimum and maximum height.
+ */
+@protocol MDCFlexibleHeaderMinMaxHeightDelegate <NSObject>
+@required
+
+/**
+ Informs the receiver that the maximum height has changed.
+ */
+- (void)flexibleHeaderMaximumHeightDidChange:(nonnull MDCFlexibleHeaderMinMaxHeight *)safeAreas;
```

Comment:
_Question:_ Why is there nothing for `-(void)flexibleHeaderMinimumHeightDidChange`?

Reply:
Similar to the answer above: the original code did not need it.